pkgs/applications/editors/eclipse/plugins.nix

```diff
@@ -447,6 +447,27 @@ rec {
     };
   };
 
+  vrapper = buildEclipseUpdateSite rec {
+    name = "vrapper-${version}";
+    version = "0.72.0";
+    owner = "vrapper";
+    repo = "vrapper";
+    date = "20170311";
+
+    src = fetchzip {
```

Comment:
Why not `fetchFromGitHub`?

Reply:
I believe this has to be `fetchzip` because it needs to download the generated update site, not the actual repository content.

Reply:
Oh right interesting, my bad.

Reply:
Mainly this was done as there is no other plugin currently using fetchFromGithub. I'm happy to use it if that's the preferred way.

Reply:
@rycee is correct, the generated update site is required rather than the repo content so I've left that as is.

Reply:
@rycee @stumoss If you don't mind, what do you mean by the generated update site? (I'm not disputing just curious.)

Reply:
@vyp I'm not an expert on eclipse plugins (I just had an itch for vim compatible keybindings) but as far as I understand an update site is just the "Eclipse way" of packaging plugins. Essentially you generate a directory structure and xml file that describes the structure. This can then be pointed to from within eclipse's "Install New Software" menu. Eclipse will know how to install the software assuming you have packaged it correctly.

I believe [this link](https://help.eclipse.org/neon/index.jsp?topic=%2Forg.eclipse.pde.doc.user%2Fconcepts%2Fupdate_site.htm) might help describe it better than I can.

Reply:
@vyp Eclipse has this concept of an "update site" that it uses to download plugins (and check for updates). This is just a static web site having a special directory structure holding one or more plugins (in Java JAR files) and some meta files. For vrapper the update site seems to be here: http://vrapper.sourceforge.net/update-site/stable/ and the root file is typically called [site.xml](http://vrapper.sourceforge.net/update-site/stable/site.xml). To install plugins through Eclipse you can add the update site URL and Eclipse will let you install any plugins found there.

Fortunately for us, many plugin authors also provide the update site as a downloadable ZIP file and the support code in [plugins.nix](https://github.com/NixOS/nixpkgs/blob/5562181d325cfe5eb2fcd4eca49f028f2991171a/pkgs/applications/editors/eclipse/plugins.nix#L45-L82) knows about the directory structure and can install plugins from this downloaded update site.

As a consequence we have that the Eclipse plugins in NixOS are all installed as prebuilt JARs. None of them are built from source and therefore one typically wouldn't set the derivation `src` to download the plugin source code (like using `fetchFromGitHub`).

Reply:
Wow that's super interesting, thank you both for your detailed answers!